### PR TITLE
Remove navigate to missing TeamMemberDetails screen

### LINKED ISSUE #430: Clicking on team member in TeamDetails screen crashes
Uncovered in #428 

Clicking on a Team Member from within the Team Details screen crashes. There is where withNavigation is being called. Again, not entirely sure what's happening here or what is supposed to be happening here.

image

### LINKED ISSUE #433: Crash when trying to make a new message
Invariant Violation: Picker has been removed from React Native. It can now be installed and imported from '@react-native-picker/picker' instead of 'react-native'. See https://github.com/react-native-picker/picker

Introduced in #428 or #426

### PR DESCRIPTION
**Pull request by justinpchang**
- Fix #430 
- Fix #433

### CODE REVIEW

File: screens/message-details-screen/index.js [hunk 1]
```diff
@@ -17,61 +17,74 @@ type PropsType = {
...
 
-const MessageDetailsScreen = ({ messages, navigation, teams }: PropsType): React$Element<View> => {
-
-    const message = messages[navigation.state.params.messageId];
+const MessageDetailsScreen = ({ messages, navigation, teams, route }: PropsType): React$Element<View> => {
+    const message = messages[route.params.messageId];
     const teamId = message.teamId;
     const team = teams[teamId] || {};
     return (
-        <SafeAreaView style={ styles.container }>
```
Comment (justinpchang) [inline screens/message-details-screen/index.js:26]: No changes on this file from here down. Just prettier

File: screens/message-details-screen/index.js [hunk 2]
```diff
@@ -17,61 +17,74 @@ type PropsType = {
     teams: Object
 };
 
-const MessageDetailsScreen = ({ messages, navigation, teams }: PropsType): React$Element<View> => {
-
-    const message = messages[navigation.state.params.messageId];
+const MessageDetailsScreen = ({ messages, navigation, teams, route }: PropsType): React$Element<View> => {
+    const message = messages[route.params.messageId];
```
Comment (justinpchang) [inline screens/message-details-screen/index.js:21]: `navigation.state` was deprecated to `route`

File: screens/message-summaries-screen/index.js [hunk 3]
```diff
@@ -117,25 +103,28 @@ type MessagePropsType = {
...
                     actions.selectTeamById(message.teamId);
                     navigation.navigate("TeamDetails");
                 };
-            case messageTypes.REQUEST_TO_JOIN :
+            case messageTypes.REQUEST_TO_JOIN:
                 return () => {
                     const membershipId = ((message.sender || {}).email || {}).uid || "";
                     const teamId = message.teamId;
                     actions.readMessage(message, userId);
-                    navigation.navigate("TeamMemberDetails", { teamId, membershipId });
```
Comment (justinpchang) [inline screens/message-summaries-screen/index.js:136]: Only change in this file is removing the outdated navigation here

File: screens/new-message-screen/index.js [hunk 4]
```diff
@@ -1,11 +1,8 @@
 // @flow
 
 import React, { useState, useEffect, Fragment } from "react";
-import {
-    Picker,
-    ScrollView,
-    StyleSheet,
-} from "react-native";
+import { ScrollView, StyleSheet } from "react-native";
+import { Picker } from "@react-native-picker/picker";
```
Comment (justinpchang) [inline screens/new-message-screen/index.js:5]: Change was Picker was moved to its own package

File: screens/team-details-screen/index.js [hunk 5]
```diff
@@ -124,10 +124,6 @@ const TeamDetailsScreen = ({
         navigation.navigate("Home");
     };
 
-    const toMemberDetails = (teamId: string, membershipId: string) => {
-        navigation.navigate("TeamMemberDetails", { teamId, membershipId });
```
Comment (justinpchang) [inline screens/team-details-screen/index.js:128]: Remove outdated navigation

File: screens/trash-map-bag-tagger-screen/index.js [hunk 6]
```diff
@@ -36,124 +31,120 @@ type PropsType = {
...
+    trashCollectionSites,
+    trashDrops,
+    route
+}: PropsType): React$Element<any> => {
     const [activeTab, setActiveTab] = useState(dateIsInCurrentEventWindow() ? 1 : 0);
     const navState = { index: activeTab, routes };
-    const existingDrop = navigation.getParam("existingDrop", null);
-
-
+    const existingDrop = route.params?.existingDrop ?? null;
```
Comment (justinpchang) [inline screens/trash-map-bag-tagger-screen/index.js:49]: `getParams` was similarly deprecated for `route`

**Review by nfloersch — APPROVED** [review]

**Pull request merged**